source/georef/street_network.cpp

```diff
@@ -573,8 +575,8 @@ Path create_path(const GeoRef& geo_ref,
         last_transport_carac = transport_carac;
         path_item.way_idx = edge.way_idx;
         path_item.transportation = transport_carac;
-        path_item.duration += edge.duration;
-        p.duration += edge.duration;
+        path_item.duration += edge.duration / speed_factor;
+        p.duration += path_item.duration;
```

Comment:
I don't understand your comment. The way I see it: we now compute a correct `path_item.duration` first (dealing with speed factor), then add this `path_item.duration` to path duration.
Would have been wrong to add `edge.duration`, yes, but that's not what we do.